src/webview/create-component/createComponentLoader.ts

```diff
@@ -141,4 +171,111 @@ export default class CreateComponentLoader {
+                            const yaml = YAML.stringify(devfileV2, { sortMapEntries: true });
+                            fs.writeFileSync(devFileV1Path, yaml.toString(), 'utf-8');
+                            CreateComponentLoader.panel?.webview.postMessage({
+                                action: 'devfileRegenerated'
+                            });
+                        } else {
+                            closeWithMessage('Failed to delete devfile.yaml, Unable to proceed the component creation', uri);
+                        }
+                    } catch (e) {
+                        closeWithMessage('Failed to parse devfile v1, Unable to proceed the component creation', uri);
+                    }
+                } else {
+                    closeWithMessage('Devfile version not supported, Unable to proceed the component creation', uri);
+                }
+            } else {
+                compDescriptions = getCompDescription(analyzeRes);
+            }
+        } finally {
+            const devfileRegistry: DevfileRegistry[] = CreateComponentLoader.getDevfileRegistries();
+            const devfile: Devfile = devfileRegistry[0].devfiles.find((devfile) => devfile.name === compDescriptions[0].displayName);
+            void CreateComponentLoader.panel.webview.postMessage({
+                action: 'recommendedDevfile',
+                data: devfile
+            });
+        }
+    }
+
+}
+
+function closeWithMessage(title: string, folderUri: vscode.Uri) {
+    vscode.window.showErrorMessage(title);
+    close({
+        folder: folderUri
+    });
+}
+
+async function close(event: any) {
+    try {
+        const selctedFolder: vscode.Uri = vscode.Uri.from(event.folder);
+        if (childProcess && event.notification !== 'Component created') {
+            treeKill(childProcess.pid, async () => {
+                await deleteDirectory(selctedFolder.fsPath);
+            });
+        }
+    } catch (error) {
+        vscode.window.showErrorMessage('Error occurred while killing the repository clone process');
+    }
+    panel?.dispose();
+}
+
+function deleteDirectory(dir: string) {
+    return new Promise<void>(function (_resolve, reject) {
+        fs.rmdir(dir, { recursive: true }, err => {
+            if (err) {
+                reject(err);
+            }
+        })
+    });
+};
+
+function getCompDescription(devfiles: AnalyzeResponse[]): ComponentTypeDescription[] {
+    const compDescriptions = ComponentTypesView.instance.getCompDescriptions();
+    if (devfiles.length === 0) {
+        return Array.from(compDescriptions);
+    }
+    return Array.from(compDescriptions).filter(({ name, version, registry }) => devfiles.some((res) => res.devfile === name &&
+        res.devfileVersion === version && res.devfileRegistry === registry.name));
+}
+
+function deleteFile(file: string): Promise<boolean> {
+    return new Promise<boolean>(function (resolve, _reject) {
+        try {
+            fs.unlinkSync(file)
```

Comment:
if you import `fs/promises` as `fs`, then there is probably a version of `unlink` that returns a promise